### PR TITLE
Convert IC staging job into a proper j-job

### PR DESCRIPTION
**Pull request by AnilKumar-NOAA**
**Description**
Convert the job used to stage initial conditions for forecast-only into a full j-job/exscript.

**Type of change**
* create a j-job `JGLOBAL_STAGE_IC` and ex-script exglobal_stage_ic.sh to allow staging of initial conditions for GFS and GEFS forecast-only experiments.
* `coupled_ic.sh` (rename to `stage_ic.sh`) should be the rocoto wrapper on `JGLOBAL_STAGE_IC`.

The initial conditions should be ready to go -- i.e. all pre-processing has been done prior to staging.

***Acceptance Criteria (Definition of Done)**
Call sequence:
`stage_ic.sh -> JGLOBAL_STAGE_IC -> exglobal_stage_ic.sh`

**Dependencies**
May need an update in the files situated under BASE_CPLIC that serves as a repository for ICs.

### CODE REVIEW

**Comment by AnilKumar-NOAA** [comment]
Still working on stage_ic.sh. Trying to understand block by block the operations. 
These two statements need help with in stage_ic.sh - create dir and copy the IC conditions.
mkdir ${COMROT}/${PSLOT}/gefs.20210323
cp -as /scratch1/NCEPDEV/global/glopara/data/ICSDIR/C48C48mx500/gefs.20210323 ${COMROT}/${PSLOT}/gefs.20210323

**Review by aerorahul — COMMENTED** [review]
I think you need to move `configs/gfs/config.coupled_ic`  to `config.stage_ic`
Otherwise this looks good and should be testable for the gfs + forecast-only

File: scripts/exglobal_stage_ic.sh [hunk 1]
```diff
@@ -36,16 +36,16 @@ source="${BASE_CPLIC}/${CPL_ATMIC}/${PDY}${cyc}/${CDUMP}/${CASE}/INPUT/gfs_ctrl.
 target="${COM_ATMOS_INPUT}/gfs_ctrl.nc"
 ${NCP} "${source}" "${target}"
 rc=$?
-[[ ${rc} -ne 0 ]] && error_message "${source}" "${target}" "${rc}"
-err=$((err + rc))
+(( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
+(( err = err + rc ))
```
Comment (aerorahul) [inline scripts/exglobal_stage_ic.sh]: I think @WalterKolczynski-NOAA would like us to revert this to
```suggestion
err=$((err + rc))
```

File: scripts/exglobal_stage_ic.sh [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+export job="stage_ic"
+export jobid="${job}.$$"
+
+# Execute the JJOB
+
+source "${HOMEgfs}/ush/jjob_header.sh" -e "stage_ic" -c "base stage_ic"
+
+# Locally scoped variables and functions
+GDATE=$(date -d "${PDY} ${cyc} - ${assim_freq} hours" +%Y%m%d%H)
```
[the remark this answers is not in the document]
Comment (aerorahul) [reply scripts/exglobal_stage_ic.sh:7]: add a comment before this line too ignore the possible misspelling (it is not)
```
# shellcheck disable=SC2153
```

[the remark this answers is not in the document]
Comment (AnilKumar-NOAA) [reply scripts/exglobal_stage_ic.sh:7]: Added Shell check disable command for code warning SC2153.

**Review by AnilKumar-NOAA — COMMENTED** [review]
Done

**Comment by WalterKolczynski-NOAA** [comment]
I'll review this tomorrow when I 'm back from leave, but in the meantime can you fix the title and description?

**Comment by TerrenceMcGuinness-NOAA** [comment]
My account running C48_ATM does not have permission to `${DATA}`.  I did not see it set in it's **EXPDIR** and at this point still can not determine its intended value.

It looks like its "intended" value set at runtime in [jjobs_header.sh](https://github.com/NOAA-EMC/global-workflow/blob/573ecce89262bb8d11e7ce2808742b7f1d9be938/ush/jjob_header.sh#L69)

```
mterry (Orion-login-3) PR $ grep DATA /work2/noaa/stmp/GFS_CI_ROOT/PR/1806/global-workflow/jobs/JGLOBAL_STAGE_IC
cd "${DATAROOT}" || (echo "${DATAROOT} does not exist. ABORT!"; exit 1)
[[ ${KEEPDATA} = "NO" ]] && rm -rf "${DATA}"
```


Here looks like part of the issue ....

The dir for **DATAROOT** was never created:

```
(gfs_workflow) mterry (Orion-login-3) C48_ATM_64d65eb3 $ grep DATAROOT C48_ATM_64d65eb3.xml | head -1
	<envar><name>DATAROOT</name><value>/work/noaa/stmp/mterry/RUNDIRS/C48_ATM_64d65eb3</value></envar>

(gfs_workflow) mterry (Orion-login-3) C48_ATM_64d65eb3 $ ls -1rtd /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_64d65eb3
ls: cannot access /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_64d65eb3: No such file or directory

(gfs_workflow) mterry (Orion-login-3) C48_ATM_64d65eb3 $ ls -1rtd /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_* | tail
/work/noaa/stmp/mterry/RUNDIRS/C48_ATM_1a83e794
/work/noaa/stmp/mterry/RUNDIRS/C48_ATM_114512a8
/work/noaa/stmp/mterry/RUNDIRS/C48_ATM_c2ad7e0e
/work/noaa/stmp/mterry/RUNDIRS/C48_ATM_eb1d05a9
/work/noaa/stmp/mterry/RUNDIRS/C48_ATM_307abf8e
/work/noaa/stmp/mterry/RUNDIRS/C48_ATM_ade24c29
/work/noaa/stmp/mterry/RUNDIRS/C48_ATM_2974811c
/work/noaa/stmp/mterry/RUNDIRS/C48_ATM_c4930f30
/work/noaa/stmp/mterry/RUNDIRS/C48_ATM_d3f31fae
/work/noaa/stmp/mterry/RUNDIRS/C48_ATM_5a4e4de6
(gfs_workflow) mterry (Orion-login-3) C48_ATM_64d65eb3 $

**Comment by AnilKumar-NOAA** [comment]
> My account running C48_ATM does not have permission to `${DATA}`. I did not see it set in it's **EXPDIR** and at this point still can not determine its intended value.
> 
> It looks like its "intended" value set at runtime in [jjobs_header.sh](https://github.com/NOAA-EMC/global-workflow/blob/573ecce89262bb8d11e7ce2808742b7f1d9be938/ush/jjob_header.sh#L69)
> 
> ```
> mterry (Orion-login-3) PR $ grep DATA /work2/noaa/stmp/GFS_CI_ROOT/PR/1806/global-workflow/jobs/JGLOBAL_STAGE_IC
> cd "${DATAROOT}" || (echo "${DATAROOT} does not exist. ABORT!"; exit 1)
> [[ ${KEEPDATA} = "NO" ]] && rm -rf "${DATA}"
> ```
> 
> Here looks like part of the issue ....
> 
> The dir for **DATAROOT** was never created:
> 
> ```
> (gfs_workflow) mterry (Orion-login-3) C48_ATM_64d65eb3 $ grep DATAROOT C48_ATM_64d65eb3.xml | head -1
> 	<envar><name>DATAROOT</name><value>/work/noaa/stmp/mterry/RUNDIRS/C48_ATM_64d65eb3</value></envar>
> 
> (gfs_workflow) mterry (Orion-login-3) C48_ATM_64d65eb3 $ ls -1rtd /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_64d65eb3
> ls: cannot access /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_64d65eb3: No such file or directory
> 
> (gfs_workflow) mterry (Orion-login-3) C48_ATM_64d65eb3 $ ls -1rtd /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_* | tail
> /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_1a83e794
> /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_114512a8
> /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_c2ad7e0e
> /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_eb1d05a9
> /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_307abf8e
> /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_ade24c29
> /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_2974811c
> /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_c4930f30
> /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_d3f31fae
> /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_5a4e4de6
> (gfs_workflow) mterry (Orion-login-3) C48_ATM_64d65eb3 $
> ```

Terry - what should be done next? shall we run again?

**Comment by AnilKumar-NOAA** [comment]
Terry - what should be done next? shall we run again?

File: scripts/exglobal_stage_ic.sh [hunk 3]
```diff
@@ -2,19 +2,11 @@
...
-status=$?
-[[ ${status} -ne 0 ]] && exit ${status}
 
-export job="coupled_ic"
-export jobid="${job}.$$"
-
-# Execute the JJOB
-
-source "${HOMEgfs}/ush/jjob_header.sh" -e "coupled_ic" -c "base coupled_ic"
+source "${HOMEgfs}/ush/jjob_header.sh" -e "stage_ic" -c "base stage_ic"
```
Comment (aerorahul) [inline scripts/exglobal_stage_ic.sh]: This was already done in the j-job.  So, it does not belong here.

**Review by aerorahul — CHANGES_REQUESTED** [review]
I think your problems might be from the fact that you have a duplicate line in your ex-script that doesn't belong there.

File: scripts/exglobal_stage_ic.sh (hunk 3 above)
Comment (AnilKumar-NOAA) [reply scripts/exglobal_stage_ic.sh]: I'll take out from exglobal_stage file then. Yes also there in j-job. let me check and run again

File: jobs/rocoto/stage_ic.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+# Source FV3GFS workflow modules
+. "${HOMEgfs}/ush/load_fv3gfs_modules.sh"
+status=$?
+[[ "${status}" -ne 0 ]] && exit "${status}"
+
+export job="stage_ic"
+export jobid="${job}.$$"
+
+# Execute the JJOB
+"${HOMEgfs}"/jobs/JGLOBAL_STAGE_IC
```
Comment (WalterKolczynski-NOAA) [inline jobs/rocoto/stage_ic.sh]:
```suggestion
"${HOMEgfs}/jobs/JGLOBAL_STAGE_IC"
```

File: ush/load_fv3gfs_modules.sh [hunk 5]
```diff
@@ -1,7 +1,7 @@
 #! /usr/bin/env bash
 
 ###############################################################
-if [[ "${DEBUG_WORKFLOW:-NO}" == "NO" ]]; then
```
Comment (WalterKolczynski-NOAA) [inline ush/load_fv3gfs_modules.sh]: This is incorrect. Please undo this change.

File: global-workflow
Comment (WalterKolczynski-NOAA) [inline global-workflow:1]: What is this submodule?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: ush/load_fv3gfs_modules.sh [hunk 6]
```diff
@@ -1,7 +1,7 @@
...
+if [[ "${DEBUG_WORKFLOW:-NO}" == "YES" ]]; then
```
Comment (aerorahul) [inline ush/load_fv3gfs_modules.sh]:
```suggestion
if [[ "${DEBUG_WORKFLOW:-NO}" == "NO" ]]; then
```

File: ush/load_fv3gfs_modules.sh (hunk 5 above)
Comment (AnilKumar-NOAA) [reply ush/load_fv3gfs_modules.sh]: I undo this change.

**Comment by WalterKolczynski-NOAA** [comment]
@AnilKumar-NOAA While we wait for CI testing to complete, would you please update the title to say what was done?

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Review by aerorahul — APPROVED** [review]
Looks good.
Thank you
